fetch: Allow to change the local dest path to a direct path

fclairamb opened the pull request:
Allows the "fetch" module to change the local destination file or directory to anything we like and not just $host/$source_path .

It doesn't break the compatibility. It adds an optional dest_prefix parameter.

There's no reason for "fetch" to be forced to reproduce the remote path.

Commit:
Adds a dest_prefix option with :
  - its default value to 'host_source_path' being the current default bevior
  - a new value of 'no' to disable it

mpdehaan commented:
Ok, I understand I think what you are trying to do, but the documentation is not sufficient to understand how to use this.

Can you please provide more examples and explanation?

Commit:
Changed the dest_prefix possible values and added documentation

* I realized the previous dest_prefix=no value was breaking the idea
of having dest as a directory only (as stated in the documentation).
It can now be used with dest_prefix=simple, which only use the dest
option directory and the filename without its path.
Which means we can't chose the destination filename anymore, but it's
not disturbing in my use cases.
Using the dest_prefix=simple option with more than one host means that
you have to define the dest and/or src options with non static values
if you don't want to overwrite the same file.

* Updated the documentation

fclairamb commented:
Ok, so I changed the code to respect the "dest" definition and updated the documentation accordingly.

### Short examples

Default behavior (unchanged):

```
$ ansible mygroup -u root -m fetch -a "src=/etc/profile dest=/tmp"
1.1.1.1 | success >> {
    "changed": true, 
    "dest": "/tmp/1.1.1.1/etc/profile", 
    "md5sum": "b94c2e3df2a779ac12080942df4d86ea"
}
```

Default behavior by specificying the dest_prefix option:

```
$ ansible mygroup -u root -m fetch -a "src=/etc/profile dest=/tmp dest_prefix=host_source_path"
1.1.1.1 | success >> {
    "changed": false, 
    "dest": "/tmp/1.1.1.1/etc/profile", 
    "file": "/etc/profile", 
    "md5sum": "b94c2e3df2a779ac12080942df4d86ea"
}
```

dest_prefix=host example:

```
$ ansible mygroup -u root -m fetch -a "src=/etc/profile dest=/tmp dest_prefix=host"
1.1.1.1 | success >> {
    "changed": true, 
    "dest": "/tmp/1.1.1.1/profile", 
    "md5sum": "b94c2e3df2a779ac12080942df4d86ea"
}
```

dest_prefix=simple example:

```
$ ansible mygroup -u root -m fetch -a "src=/etc/profile dest=/tmp dest_prefix=simple"
1.1.1.1 | success >> {
    "changed": true, 
    "dest": "/tmp/profile", 
    "md5sum": "b94c2e3df2a779ac12080942df4d86ea"
}
```

mpdehaan commented:
So the doc example mentions "profile", but not where "profile" comes from.  I assume is the end of the filename.  Can you update the docs so this is more clear and also include an example?

Commit:
Updated the documentation to be a little bit more explicit.

mpdehaan commented:
I have decided to not take this one on the basis that this is a very confusing paragraph still and it is easy enough to get the remote path out of the fetch module.

fclairamb commented:
Ok I understand. I'll try to be more clear next time. And for now, I'll use some "local_action".
